catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswEndpoint.java

```diff
@@ -1292,10 +1307,22 @@ private boolean isProductRetrieval(List<String> ids, String outputFormat, String
         return false;
     }
 
-    private CswRecordCollection queryProductById(String id) throws CswException {
+    private CswRecordCollection queryProductById(String id, String rangeValue)
+            throws CswException, UnsupportedQueryException {
 
         final ResourceRequestById resourceRequest = new ResourceRequestById(id);
 
+        if (StringUtils.isNotBlank(rangeValue)) {
+            String range = getRange(rangeValue);
+            if (range != null) {
+                long bytesToSkip = Long.parseLong(range);
```

Comment:
Your `getRange` method implies to me that a valid range _might_ be `10-899` (which certainly makes semantic sense as a range) but in that case you would throw an exception trying to parse it here.